app/models/user.rb

```diff
@@ -1,24 +1,55 @@
 class User < ApplicationRecord
   PERMITTED_PARAMS = %i(name email password password_validation).freeze
 
-  validates :name,
-            presence: true,
+  attr_accessor :remember_token
+
+  validates :name, presence: true,
             length: {maximum: Settings.validate.user.name_maxlength}
 
-  validates :email,
-            presence: true,
+  validates :email, presence: true,
             length: {maximum: Settings.validate.user.email_maxlength},
             format: {with: Settings.validate.user.email_format},
             uniqueness: {case_sensitive: false}
 
-  validates :password,
-            presence: true,
+  validates :password, presence: true,
             length: {minimum: Settings.validate.user.password_minlength}
 
   has_secure_password
 
   before_save :email_downcase
 
+  class << self
+    def digest string
```

Comment:
Hàm này digest string như thế nào?

Reply:
Hàm này hash string sử dụng bcrypt theo param cost, cost càng cao thì digest càng tốn tài nguyên để decrypt, tuy vậy chi phí tính toán để tạo hash cũng cao hơn. Hàm sử dụng param cost tối thiểu để hash trong môi trường test và normal cost trong production.